### PR TITLE
Avoid the crash with the audio which contains only silence

### LINKED ISSUE #59: Bug with trimming
I train the network with default hyper-parameters:

`Exception in thread Thread-2:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 810, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 763, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/jesse/tensorflow_workspace/tensorflow-wavenet/audio_reader.py", line 87, in thread_main
    audio = trim_silence(audio[:, 0])
  File "/home/jesse/tensorflow_workspace/tensorflow-wavenet/audio_reader.py", line 47, in trim_silence
    return audio[indices[0]:indices[-1]]
IndexError: index 0 is out of bounds for axis 0 with size 0`

The error occurs at the last line of trim_silence method:

`return audio[indices[0]:indices[-1]]`

It seems that some audio consists of only silence and is blank after trimming.

### PR DESCRIPTION
**Pull request by mecab**
Also add warning output for the silence.

This will fix #59.

### CODE REVIEW

**Comment by jyegerlehner** [comment]
Hi mecab,
I get the following exception when I run this:
```
Exception in thread Thread-2:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 810, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 763, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/jd/dev/models/tensorflow/tensorflow-wavenet/audio_reader.py", line 94, in thread_main
    audio = trim_silence(audio[:, 0], filename)
  File "/home/jd/dev/models/tensorflow/tensorflow-wavenet/audio_reader.py", line 48, in trim_silence
    if indices:
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()

```

Yes, I'm one of those barbarians who runs python 2.7.

**Comment by mecab** [comment]
@jyegerlehner 
Thanks for pointing it out, i'll fix it.
(It happen also in python3... I pushed wrong branch.)

**Comment by mecab** [comment]
fixed.

**Comment by lemonzi** [comment]
I would print the warning in `AudioReader`; `trim_silence` doesn't need to know about filenames.

File: audio_reader.py [hunk 1]
```diff
@@ -39,12 +39,19 @@ def load_vctk_audio(directory, sample_rate):
         yield audio, speaker_id
 
 
-def trim_silence(audio, threshold=0.3):
+def trim_silence(audio, filename, threshold=0.3):
```
Comment (ibab) [inline audio_reader.py]: It was kind of nice that the `trim_silence` function  could be used on arbitrary waveforms, without supplying a filename.
We can keep the check for an empty shape, and create the warning message at the point below where `trim_silence`, then the file name wouldn't be needed here.

File: audio_reader.py [hunk 2]
```diff
@@ -78,13 +85,13 @@ def thread_main(self, sess):
...
             iterator = load_generic_audio(self.audio_dir, self.sample_rate)
-            for audio in iterator:
+            for audio, filename in iterator:
                 if self.coord.should_stop():
                     self.stop_threads()
                     stop = True
                     break
                 # Remove silence
-                audio = trim_silence(audio[:, 0])
+                audio = trim_silence(audio[:, 0], filename)
```
Comment (ibab) [inline audio_reader.py]: We can replace this with something like
```python
audio = trim_silence(audio[:, 0])
if not audio:
    # Print warning message
```

**Comment by ibab** [comment]
Thanks for fixing this!
I've made some comments on how you could implement @lemonzi's suggestion.

**Comment by mecab** [comment]
Yeah, I like it, too. I've fixed it.

**Comment by ibab** [comment]
Nice work! Merging this now.

**Pull request merged**